frontend/packages/console-shared/src/hooks/perspectiveUtils.ts

```diff
@@ -1,7 +1,125 @@
+    (await Promise.all(requiredPromises)
+      .then((values) => {
+        // enable the prespective if all the access review checks are successful
+        hasAccess = hasAccess && values.every((val) => val.status.allowed);
+      })
+      .catch((e) => {
+        // eslint-disable-next-line no-console
+        console.warn('AccessReview check failed', e);
+        hasAccess = hasAccess && false;
+      }));
+
+  missingPromises?.length > 0 &&
+    (await Promise.all(missingPromises)
+      .then((values) => {
+        // enable perspective if atleast one access review check fails
+        hasAccess = hasAccess && !values.every((val) => val.status.allowed);
+      })
+      .catch((e) => {
+        // eslint-disable-next-line no-console
+        console.warn('AccessReview check failed', e);
+        hasAccess = hasAccess && true;
+      }));
+
+  return hasAccess;
+};
+
+export const useGetPerspectives = (): LoadedExtension<PerspectiveExtension>[] => {
+  const perspectiveExtensions = useExtensions<PerspectiveExtension>(isPerspective);
+  const [extensions, setExtensions] = React.useState<LoadedExtension<PerspectiveExtension>[]>([]);
+  const adminPerspective = perspectiveExtensions.find(
+    (perspectiveEx) => perspectiveEx.properties.id === 'admin',
+  );
+
+  const setResults = React.useCallback(
+    (newResult: LoadedExtension<PerspectiveExtension>) => {
+      setExtensions((oldResults: LoadedExtension<PerspectiveExtension>[]) => {
+        const arr = [...oldResults, newResult];
+        return _.uniqBy(arr, 'properties.id');
+      });
+    },
+    [setExtensions],
+  );
+  React.useEffect(() => {
+    if (!window.SERVER_FLAGS.perspectives) {
+      setExtensions(perspectiveExtensions);
+    } else {
+      const perspectives: Perspective[] = JSON.parse(window.SERVER_FLAGS.perspectives);
+      perspectiveExtensions.forEach((perspectiveExtension) => {
+        const perspective = perspectives?.find((p) => p.id === perspectiveExtension.properties.id);
+
+        if (perspective?.visibility?.state === PerspectiveVisibilityState.Enabled) {
+          setResults(perspectiveExtension);
+        } else if (
+          perspective?.visibility?.state === PerspectiveVisibilityState.AccessReview &&
+          perspective?.visibility?.accessReview &&
+          Object.keys(perspective?.visibility?.accessReview)?.length > 0
+        ) {
+          hasReviewAccess(perspective?.visibility?.accessReview)
+            .then((res) => {
+              if (res) {
+                setResults(perspectiveExtension);
+              }
+            })
+            .catch((e) => {
+              setResults(perspectiveExtension);
+              // eslint-disable-next-line no-console
+              console.warn('Could not check access', e);
+            });
+        }
+      });
+    }
+  }, [perspectiveExtensions, setResults]);
+  const result = extensions.filter((res) => !!res);
+  return result.length === 0 ? [adminPerspective] : result;
```

Comment:
I'm not sure if this makes an issue, but in many cases returning a new array with each rendering causes performance issues later.

Can you set `[adminPerspective]` in the `setResults` instead?